packages/components/src/FilterDateRangePicker/subcomponents/FilterDateRangePickerField/FilterDateRangePickerField.tsx

```diff
@@ -0,0 +1,248 @@
+      endDateInputValue: inputRangeEndValue,
+      startDate: selectedRange?.from,
+      startDateFieldLabel: inputRangeStartLabel,
+    })
+
+  const inputRangeStartHandlers = useDateInputHandlers({
+    locale,
+    disabledDays,
+    setInputValue: setInputRangeStartValue,
+    onDateChange: date => {
+      const newDate = validateStartDate(date)
+      const endDate = parseDateFromTextFormatValue(inputRangeEndValue, locale)
+
+      if (newDate && !isInvalidDate(endDate)) {
+        const newEndDate = endDateValidation.validateEndDateBeforeStartDate({
+          startDate: newDate,
+          startDateFieldLabel: inputRangeStartLabel,
+          endDate,
+          endDateInputValue: inputRangeEndValue,
+        })
+        handleDateRangeChange({ from: newDate, to: newEndDate })
+      } else {
+        handleDateRangeChange({ from: newDate, to: rangeEnd })
+      }
+
+      if (newDate) setStartMonth(newDate)
+    },
+    ...inputRangeStartProps,
+  })
+
+  const inputRangeEndHandlers = useDateInputHandlers({
+    locale,
+    disabledDays,
+    setInputValue: setInputRangeEndValue,
+    onDateChange: date => {
+      const newDate = validateEndDate(date)
+      handleDateRangeChange({ from: rangeStart, to: newDate })
+    },
+    ...inputRangeEndProps,
+  })
+
+  const handleCalendarSelectRange: CalendarRangeProps["onSelect"] = range => {
+    const newStartDate = validateStartDate(range?.from)
+    const newEndDate = validateEndDate(range?.to)
+    setInputRangeStartValue(transformDateToInputValue(newStartDate))
+    setInputRangeEndValue(transformDateToInputValue(newEndDate))
+    handleDateRangeChange({ from: newStartDate, to: newEndDate })
+  }
+
+  useEffect(() => {
+    const newStartDate = validateStartDate(selectedRange?.from)
+    const newEndDate = validateEndDate(selectedRange?.to)
+    handleDateRangeChange({ from: newStartDate, to: newEndDate })
+  }, [])
+
+  return (
+    <div
+      className={classnames(
+        styles.filterDateRangePickerField,
+        classNameOverride
+      )}
+      {...restProps}
+    >
+      <DateRangeInputField
+        id={`${id}--input`}
+        legend={label}
+        inputRangeStartProps={{
+          labelText: inputRangeStartLabel,
+          value: inputRangeStartValue,
+          ...inputRangeStartProps,
+          // The below props extend the values from inputRangeStartProps, therefore must be below the spread
+          ...inputRangeStartHandlers,
+        }}
+        inputRangeEndProps={{
+          labelText: inputRangeEndLabel,
+          value: inputRangeEndValue,
+          ...inputRangeEndProps,
+          // The below props extend the values from inputRangeEndProps, therefore must be below the spread
+          ...inputRangeEndHandlers,
+        }}
+        locale={locale}
+        description={description}
+        validationMessage={{
+          dateStart: startDateValidation.validationMessage,
+          dateEnd: endDateValidation.validationMessage,
```

Comment:
Could you match the terminology up between `dateStart` and `startDateValidation` and have it `dateStart.validationMessage`?